static/src/javascripts/projects/commercial/modules/article-body-adverts.ts

```diff
@@ -1,66 +1,66 @@
+import type { AdSize } from '@guardian/commercial-core';
+import { adSizes } from '@guardian/commercial-core';
+import { getBreakpoint, getViewport } from 'lib/detect-viewport';
 import config from '../../../lib/config';
-import fastdom from '../../../lib/fastdom-promise';
 import { mediator } from '../../../lib/mediator';
 import { spaceFiller } from '../../common/modules/article/space-filler';
-import { adSizes } from '@guardian/commercial-core';
-import { addSlot } from './dfp/add-slot';
-import { trackAdRender } from './dfp/track-ad-render';
-import { createAdSlot } from './dfp/create-slot';
-
 import { commercialFeatures } from '../../common/modules/commercial/commercial-features';
 import { initCarrot } from './carrot-traffic-driver';
-import { getBreakpoint, getTweakpoint, getViewport } from 'lib/detect-viewport';
+import { addSlot } from './dfp/add-slot';
+import { createAdSlot } from './dfp/create-slot';
+import { trackAdRender } from './dfp/track-ad-render';
 
-const isPaidContent = config.get('page.isPaidContent', false);
+const isPaidContent = config.get<boolean>('page.isPaidContent', false);
 
 const adSlotClassSelectorSizes = {
 	minAbove: 500,
 	minBelow: 500,
 };
 
-const insertAdAtPara = (para, name, type, classes, sizes) => {
+const insertAdAtPara = (
+	para: Node,
+	name: string,
+	type: string,
+	classes?: string,
+	sizes?: Record<string, AdSize[]>,
+): void => {
 	const ad = createAdSlot(type, {
 		name,
 		classes,
 		sizes,
 	});
-
-	return fastdom
-		.mutate(() => {
-			if (para.parentNode) {
-				para.parentNode.insertBefore(ad, para);
-			}
-		})
-		.then(() => {
-			const shouldForceDisplay = ['im', 'carrot'].includes(name);
-			addSlot(ad, shouldForceDisplay);
-		});
+	if (para.parentNode) {
+		para.parentNode.insertBefore(ad, para);
+	}
+	const shouldForceDisplay = ['im', 'carrot'].includes(name);
+	addSlot(ad, shouldForceDisplay);
 };
 
-let previousAllowedCandidate;
+let previousAllowedCandidate: SpacefinderItem;
 
 // this facilitates a second filtering, now taking into account the candidates' position/size relative to the other candidates
-const filterNearbyCandidates = (maximumAdHeight) => (candidate) => {
-	if (
-		!previousAllowedCandidate ||
-		Math.abs(candidate.top - previousAllowedCandidate.top) -
-			maximumAdHeight >=
-			adSlotClassSelectorSizes.minBelow
-	) {
-		previousAllowedCandidate = candidate;
-		return true;
-	}
-	return false;
-};
+const filterNearbyCandidates =
+	(maximumAdHeight: number) => (candidate: SpacefinderItem) => {
+		if (
+			!previousAllowedCandidate ||
+			Math.abs(candidate.top - previousAllowedCandidate.top) -
+				maximumAdHeight >=
+				adSlotClassSelectorSizes.minBelow
+		) {
+			previousAllowedCandidate = candidate;
+			return true;
+		}
+		return false;
+	};
 
-const isDotcomRendering = config.get('isDotcomRendering', false);
+const isDotcomRendering = config.get('isDotcomRendering', false) as boolean;
```

Comment:
Think @arelra suggested something similar above but
```suggestion
const isDotcomRendering = config.get<boolean>('isDotcomRendering', false);
```
or similarly
```suggestion
const { isDotcomRendering } = window.guardian.config;
```